frontend/src/pages-tests/__tests__/jest/edit/[taskId].spec.tsx

```diff
@@ -0,0 +1,163 @@
+import { createMockedEmbeddedApp } from "@/components/__mocks__/EmbeddedAppMock";
+import { EmbeddedAppRef } from "@/components/EmbeddedApp";
+import EditTaskPage from "@/pages/edit/[taskId]";
+import { render } from "@testing-library/react";
+import { IntlProvider } from "react-intl";
+import English from "../../../../../content/compiled-locales/en.json";
+import waitForExpect from "wait-for-expect";
+
+const renderComponent = () => (
+  <IntlProvider locale="en" messages={English}>
+    <EditTaskPage />
+  </IntlProvider>
+);
+
+jest.mock("@/components/EmbeddedApp", () => ({
+  __esModule: true,
+  _getMockComponent: jest.fn(() => {
+    throw new Error("Not implemented");
+  }),
+  get default() {
+    return this._getMockComponent();
+  },
+}));
+
+const EmbeddedAppMock = jest.requireMock("@/components/EmbeddedApp")
+  ._getMockComponent as jest.Mock;
+
+const EmbeddedApp = jest.requireActual("@/components/EmbeddedApp").default;
+
+jest.mock("@/utilities/file-from-disk", () => ({
+  readSingleFileFromDisk: jest.fn(),
+}));
+
+const readSingleFileFromDisk = jest.requireMock("@/utilities/file-from-disk")
+  .readSingleFileFromDisk as jest.Mock;
+
+jest.mock("@/utilities/download", () => ({
+  downloadBlob: jest.fn(),
+}));
+
+const downloadBlob = jest.requireMock("@/utilities/download")
+  .downloadBlob as jest.Mock;
+
+jest.mock("next/router", () => ({
+  useRouter: jest.fn(() => ({
+    query: { sessionId: "sessionId", taskId: "taskId" },
+  })),
+}));
```

Comment:
Nothing to fix here, but I'm mildly worried about brittleness seeing how many mocks there are and that we also import some originals, and knowing how badly Jest handles this sometimes 😅

Reply:
This was not super fun to write either ^^ Especially adding the possibility to switch to the original component during the test, urrgh.